functions/go/enable-gcp-services/README.md

```diff
@@ -0,0 +1,120 @@
+# enable-gcp-services
+
+## Overview
+
+<!--mdtogo:Short-->
+
+The `enable-gcp-services` function generates [GCP project service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service)
+resources from a list of services to enable [GCP APIs](https://cloud.google.com/apis) within a specified project. This allows users to succinctly define all
+the services necessary in a single resource and have tighter control over which services are enabled in a specific project.
+
+<!--mdtogo-->
+
+<!--mdtogo:Long-->
+
+## Usage
+
+`enable-gcp-services` function can be used both declaratively and imperatively.
+
+```shell
+kpt fn eval --image gcr.io/kpt-fn/enable-gcp-services:unstable
+```
+
+The `enable-gcp-services` function does the following:
+
+1. Discovers all `ProjectServiceList` custom resources in a given package.
+
+1. For each `ProjectServiceList` CR, it generates [GCP project service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service) resources as specified in the `spec.services` list.
+    * Adds all annotations defined for `ProjectServiceList` CR to each generated resource. This can be used for enabling features like
+[disable-on-destroy](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service#custom_resource_definition_properties) for generated services.
+    * Sets namespace if any defined for `ProjectServiceList` CR to each generated resource.
+    * Sets projectID if any defined for `ProjectServiceList` CR to each generated resource.
+1. Each generated [GCP project service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service) resource
+has a `blueprints.cloud.google.com/managed-by-enable-gcp-services` annotation. This annotation allows `enable-gcp-services` function to
+track generated resources for the declarative management of the generated resources. Any changes made to the generate resources will be overwritten and should be made to the `ProjectServiceList` CR instead.
+
+### `ProjectServiceList`
+
+This function only supports local-config custom resources of kind `ProjectServiceList` and can be provided using input items along with other KRM resources. Multiple `ProjectServiceList` CRs can be declared in a package.
+
+`ProjectServiceList` has the following supported parameters:
+
+```yaml
+apiVersion: blueprints.cloud.google.com/v1alpha1
+kind: ProjectServiceList
+metadata:
+  name: my-project-services
+  annotations:
+    cnrm.cloud.google.com/deletion-policy: false
+    config.kubernetes.io/local-config: true
+spec:
+  services: # list of services to generate
+    - compute.googleapis.com
+  projectID: foo # kpt-set: ${project-id}
```

Comment:
> kpt-set: ${project-id}

The setter comment is not really related here. Please drop it to reduce the chance of user confusion.

Reply:
Added per https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/568#discussion_r705523525
After re reading the readme, I agree that it might be confusing, so dropped.